docs/rfcs/rfc-029-connect-in-update-mutations.md

```diff
@@ -0,0 +1,281 @@
+    WHERE this_connect_sponsor0_node.id = "2"
+    CALL {
+        WITH *
+        WITH collect(this_connect_sponsor0_node) as connectedNodes, collect(this) as parentNodes
+        UNWIND parentNodes as this
+        UNWIND connectedNodes as this_connect_sponsor0_node
+        MERGE (this)-[this_connect_sponsor0_relationship:HAS_SPONSOR]->(this_connect_sponsor0_node) # This is the line that causes the relationship to be overwritten instead of a new one being created
+        SET this_connect_sponsor0_relationship.type = "newType"
+        SET this_connect_sponsor0_relationship.startDate = "0123-01-01"
+        RETURN count(*) AS _
+    }
+    RETURN count(*) AS connect_this_connect_sponsor_Client
+}
+WITH *
+RETURN collect(DISTINCT this { .id }) AS data
+```
+
+This would merge an existing "HAS_SPONSOR" relation, overwriting the old relation with the properties provided in the `edge` input.
+
+### Questions
+
+* What should be the default behaviour? Existing MERGE behaviour? Or always create a new relationship?
+* How would this be presented in query results? One of each, or list all of the individual related nodes?
+* How would disconnect work? Disconnect one or disconnect all?
+* How do aggregations work? Do we aggregate across all relationships, or just one?
+
+## Proposed Solution
+
+* Add the possibility to create new connections, instead of always overwriting existing relationships.
+* The existing default behaviour should be kept the same for now to avoid a breaking change.
+* It should be possible to overwrite the default behaviour when specifying the `@relationship` directive. This can be achieved with a `defaultUpdateOperation` argument that accepts either CREATE or UPDATE. This should be defaulted to UPDATE to avoid breaking changes.
+* Provide the option to the user to make use of either behaviour at query time. This can be achieved with an `operation` argument on to the `connect` input that accepts either CREATE or UPDATE. If this argument is not provided, maintain the default behaviour defined by `defaultUpdateOperation`.
+* If there are several connections of the same type, to the same nodes, these should be represented with the nodes being returned multiple times in the response. This is already the current behaviour.
+* To maintain consistency with returning duplicate nodes if multiple relationships, aggregetions should be accross all relationships. This is already the current behaviour.
+* Disconnect should disconnect all relations that meet the query filters (e.g. could use limit 1 to make it delete only a single relationship). This is already the current behaviour.
+
+### Usage Examples
+
+#### `@relationship` definition
+
+Specifying the `defaultUpdateOperation` on the `@relationship` directive:
+
+```gql
+type Client {
+    id: String!
+    login: String!
+    sponsor: [Client!]! @relationship(type: "HAS_SPONSOR", properties: "HasSponsor", direction: OUT, defaultUpdateOperation: "CREATE")
```

Comment:
I would change up the argument name here to make it clear it is for connect operations. Once again, I have played with having it as a boolean, I'm open to a discussion!

```suggestion
    sponsor: [Client!]! @relationship(type: "HAS_SPONSOR", properties: "HasSponsor", direction: OUT, connectIfExists: true)
```